lib/db.js

```diff
@@ -790,6 +791,50 @@ module.exports = function (
     return this.pool.get('/securityEvents/' + params.uid.toString('hex') + '/ip/' + params.ipAddr)
   }
 
+  DB.prototype.createUnblockCode = function (uid) {
+    log.trace({
+      op: 'DB.createUnblockCode',
+      uid: uid
+    })
+    var self = this
+    var unblock = UnblockCode()
+    return this.pool.put('/account/' + uid.toString('hex') + '/unblock/' + unblock)
+      .then(
+        function () {
+          return unblock
+        },
+        function (err) {
+          // duplicates should be super rare, but it's feasible that a
+          // uid already has en existing unblockCode. Just try again.
+          if (isRecordAlreadyExistsError(err)) {
+            log.error({
+              op: 'DB.createUnblockCode.duplicate',
+              err: err,
+              uid: uid
+            })
+            return self.createUnblockCode(uid)
```

Comment:
We could add some bookkeeping, would require cleaning up afterwards to prevent memory leaks. Something like this:

```js
createUnblockCode(uid) {
  var key = uid.toString('hex')
  if (this.createUnblockCode[key] > 3) {
    throw somethingBadHappened()
  } else if (this.createUnblockCode[key]) {
    this.createUnblockCode[key]++
  } else {
    this.createUnblockCode[key] = 1
  }
  this.pool.post().then((results) => {
    delete this.createUnblockCode[key]
  })
  // ...
}

```